include/swift/Sema/ConstraintSystem.h

```diff
@@ -5537,7 +5537,12 @@ class ConstraintSystem {
       Type wrapperType, Type paramType, ParamDecl *param, Identifier argLabel,
       ConstraintKind matchKind, ConstraintLocatorBuilder locator);
 
-  Optional<BindingSet> determineBestBindings();
+  /// Determine whether given type variable with its set of bindings is viable
+  /// to be attempted on the next step of the solver.
+  ///
+  /// \c isSingle indicates whether this is the only component produced
+  /// by "split" step, in which case bindings should not be printed in debug output.
+  Optional<BindingSet> determineBestBindings(bool isSingle = false);
```

Comment:
We talked about this offline - should switch from flag to a more general callback e.g. `llvm::function_ref<void(const BindingSet &)> onCandidate` which is going to be used for printing at each call site.